packages/web-config-server/src/apiV1/report.js

```diff
@@ -0,0 +1,123 @@
+import { CustomError, replaceValues, convertDateRangeToPeriodQueryString } from '@tupaia/utils';
+import { DataAggregatingRouteHandler } from './DataAggregatingRouteHandler';
+import { ReportPermissionChecker } from './permissions';
+import { ReportConnection } from '/connections';
+import { getDataBuilder } from '/apiV1/dataBuilders/getDataBuilder';
+import { getDhisApiInstance } from '/dhis';
+
+const reportFail = {
+  type: 'Report Error',
+  responseStatus: 400,
+};
+
+const noReportWithCode = {
+  responseText: {
+    status: 'reportError',
+    details: 'No report with corresponding code',
+  },
+};
+
+const noDataBuilder = {
+  responseText: {
+    status: 'reportError',
+    details: 'No data builder defined for current report',
+  },
+};
+
+const getIsValidDate = dateString => !Number.isNaN(Date.parse(dateString));
+
+export class ReportHandler extends DataAggregatingRouteHandler {
+  static PermissionsChecker = ReportPermissionChecker;
+
+  buildResponse = async () => {
+    const { startDate, endDate } = this.query;
+    if (getIsValidDate(startDate)) this.startDate = startDate;
+    if (getIsValidDate(endDate)) this.endDate = endDate;
+
+    const { reportCode } = this.params;
+    const { legacy } = this.query;
+
+    const reportData =
+      legacy === 'true'
+        ? await this.buildLegacyReportData(reportCode)
+        : await this.buildReportData(reportCode);
+
+    return {
+      ...reportData,
+      startDate: this.startDate,
+      endDate: this.endDate,
+    };
+  };
+
+  async buildReportData(reportCode) {
```

Comment:
I actually haven't tested this route with any new reports yet. Solutions team implemented some new `reportServerDataBuilder` dashboards but they haven't been tested. Once they're tested, I can migrate them to use the new structure to test if this is working

Reply:
Oh I have a test report I've been using, so I ran your branch to see if it works.

![image](https://user-images.githubusercontent.com/69437175/122328374-a6246300-cf83-11eb-810d-1a560113dfe6.png)